File: jdroid-gradle-commons-plugin/src/main/groovy/com/jdroid/gradle/commons/utils/ProjectUtils.java
package com.jdroid.gradle.commons.utils;

import com.jdroid.gradle.commons.BaseGradleExtension;

import org.gradle.api.Project;

public class ProjectUtils {
	
	public static <T extends BaseGradleExtension> T getJdroidExtension(Project project) {
		return ((T)project.getExtensions().getByName("jdroid"));
	}
}

File: jdroid-gradle-commons-plugin/src/main/groovy/com/jdroid/gradle/commons/utils/ProjectUtils.kt
package com.jdroid.gradle.commons.utils

import com.jdroid.gradle.commons.BaseGradleExtension

import org.gradle.api.Project

object ProjectUtils {

    @JvmStatic
    @Suppress("UNCHECKED_CAST")
    fun <T : BaseGradleExtension> getJdroidExtension(project: Project): T {
        return project.extensions.getByName("jdroid") as T
    }
}
